### ISSUE TITLE
Feature Request: add time property to truffle-config.js

### ISSUE THREAD

**Issue by pieterhartel** [trufflesuite/truffle#1891]
## Issue

`Ganache-core` and `ganache-cli` have options to set a specific time stamp on the genesis block, but there is no way to specify this in a `truffle-config.js`

## Expected Behavior

Adding a `genesis_time` property as suggested below would enable `truffle test` to start the default `ganache-core` with the appropriate time.

```
module.exports = {
 ...
      genesis_time: "2019-04-09T02:10:14Z"
 ...
};
```

**Comment by fainashalts** [comment]
Thanks for the suggestion @pieterhartel, we'll look into what this might entail!

**Comment by CruzMolina** [comment]
## Requirements

1. Enable the usage of a `genesis_time` property in a given `truffle-config.js`.

Example:
```
module.exports = {
  networks: {
    development: {
      host: "127.0.0.1",
      port: 8545,
      genesis_time: "2019-04-09T02:10:14Z",
      network_id: "*"
    }
  }
};
```

2. Throw a helpful `Error` if `genesis_time` is *not* passed a valid `Date`.
3. Ensure that the valid `genesis_time` value is passed as a configurable option internally within truffle before starting a default `ganache-core` chain for `truffle test`.
4. Add an integration mocha test to proper functionality (proper behavior when passed a valid `Date` and proper error handling when passed an invalid `Date`).

## Prerequisite Context

- Familiarity with the options `ganache-core` is expecting on its side for setting a genesis block time: https://github.com/trufflesuite/ganache-core#options
- Familiarity with the entry point for `truffle test`: https://github.com/trufflesuite/truffle/blob/develop/packages/truffle-core/lib/commands/test.js, specifically [these lines](https://github.com/trufflesuite/truffle/blob/develop/packages/truffle-core/lib/commands/test.js#L147-L167).
- Familiarity with the script for spinning up the default `ganache-core` in testing: https://github.com/trufflesuite/truffle/blob/develop/packages/truffle-core/chain.js
- Possibly familiarity with the chain environment file for error handling (might be better handled earlier in the `truffle test` sequence): https://github.com/trufflesuite/truffle/blob/develop/packages/truffle-core/lib/environment.js
- Familiarity with current `truffle test` integration tests, i.e. https://github.com/trufflesuite/truffle/blob/develop/packages/truffle/test/scenarios/happy_path/happypath.js, particularly [these lines](https://github.com/trufflesuite/truffle/blob/develop/packages/truffle/test/scenarios/happy_path/happypath.js#L113-L125).

**Comment by krsnnik** [comment]
I'd love to take a look at adding this!

**Comment by eggplantzzz** [comment]
@krsnnik Let us know if you have any questions about this. Good luck!

**Comment by gnidan** [comment]
Released the PR for this in v5.0.29! Try it out and let us know how it works for you!

Thanks! Closing this for housekeeping.
